### PR TITLE
docs improvements and updates

### PR DESCRIPTION
**Pull request by VladoLavor**

### CODE REVIEW

File: docs/intro/overview.md [hunk 1]
```diff
@@ -49,31 +49,32 @@ Another significant feature is the ability to retrieve existing VPP configuratio
...
   
 ### What It Cannot Do?
 
-The VPP agent is a control/management plane. It provides a configuration and monitoring services for the VPP dataplane. It does not decide what to do with packets arriving on any of the VPP interfaces and does not change any of the configuration parameters (routes, FIBs) based on the actual VPP state.
+The vp-agent is a control/management plane. It provides a configuration and monitoring services for the VPP dataplane. It does not decide what to do with packets arriving on any of the VPP interfaces and does not change any of the configuration parameters (routes, FIBs) based on the actual VPP state.
```
Comment (ondrej-fabry) [inline docs/intro/overview.md]: Typo `vp-agent`.

File: docs/intro/framework.md [hunk 2]
```diff
@@ -2,158 +2,106 @@
...
 
-![context][context]
+As suggested, we can view and describe the Ligato framework as a protocol Stack. See the `Ligato Architecture Stack` below. At the bottom we have the VPP and Linux kernel dataplanes. 
 
-## Design
+!!! Note 
+    While much of the Ligato development effort and software is focused on VPP, there is and certainly can be new CNFs (or even non-CNFs) that employ the linux kernel dataplane. The `one size can fit all` discussion applies here with respect to dataplane realities in the CNF universe.
 
-![VPP agent 10.000 feet][vpp-agent-10k]
+![ligato framework][ligato-framework]
```
Comment (ondrej-fabry) [inline docs/intro/framework.md]: All of these new pictures need to be a clickable and link you to the draw.io editor.

Just like this one in the Concepts section is already done: https://docs.ligato.io/en/chris/user-guide/concepts/#why-the-kv-data-store

Try clicking on the link.. it will take you to draw.io where you can edit the picture and then store in the in GitHub directly since it it's integrated together.

See the raw code for the link: https://github.com/ligato/docs/blob/master/docs/user-guide/concepts.md#L41

Basically you need to open the `*.xml` file in the draw.io directly from GitHub and then go `File -> Publish -> Link` and click `Create`.

Here are some docs explaining how to use it:
- https://github.com/jgraph/drawio/wiki/Embed-Diagrams
- https://about.draw.io/publish-link-and-embed-html/

File: docs/user-guide/concepts.md [hunk 3]
```diff
@@ -114,11 +114,11 @@ import (
...
 func New() *VPPAgent {
-	// Prepare KVDB sync plugin with ETCD as a connector
-	etcdDataSync := kvdbsync.NewPlugin(kvdbsync.UseKV(&etcd.DefaultPlugin))
+	// Prepare kvDB sync plugin with ETCD as a connector
+	etcdDataSync := kvdbsync.NewPlugin(kvdbsync.Usekv(&etcd.DefaultPlugin))
 	
-	// Put the KVDB sync to a list of proto watchers 
-	watchers := datasync.KVProtoWatchers{
+	// Put the kvDB sync to a list of proto watchers 
+	watchers := datasync.kvProtoWatchers{
```
Comment (ondrej-fabry) [inline docs/user-guide/concepts.md]: @chrismetz09 This code would not even compile.. don't change the code... and IMHO, I think `KV` is better than `kv` since it's an abbrevation of `key-value`.

Comment (chrismetz09) [reply docs/user-guide/concepts.md]: Was attempting to standardize on some term for key-value and quick scan of changes missed this. Am okay on "KV" for "kv" but need to standardize terms where it is embedded:
KVDB vs kvdb vs KVdb; 
kvdb sync vs kvdbsync vs KVDB sync;
